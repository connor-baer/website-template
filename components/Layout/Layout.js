import styled from '@emotion/styled';

import { fullWidthStyles } from '../../styles/shared';

const Layout = styled('div')(fullWidthStyles);

export default Layout;
