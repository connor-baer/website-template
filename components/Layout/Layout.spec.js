import React from 'react';

import Layout from './Layout';

describe('Layout', () => {
  function renderLayout(fn, props = {}) {
    return fn(<Layout {...props} />);
  }

  it('should render with default styles', () => {
    const children = 'label';
    const { container } = renderLayout(render, { children });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should meet basic accessibility guidelines', async () => {
    const children = 'label';
    const html = renderLayout(renderToHtml, { children });
    const result = await axe(html);
    expect(result).toHaveNoViolations();
  });
});
