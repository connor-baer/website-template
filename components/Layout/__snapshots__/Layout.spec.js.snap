// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render with default styles 1`] = `
.circuit-0 {
  max-width: 80rem;
  margin-right: auto;
  margin-left: auto;
}

@media (min-width:600px) {
  .circuit-0 {
    padding-right: 2rem;
    padding-left: 2rem;
  }
}

<div
  class="circuit-0 circuit-1"
>
  label
</div>
`;
